PGM/src/main/java/tc/oc/pgm/shop/ShopParser.java

```diff
@@ -0,0 +1,158 @@
+        }
+
+        boolean crossTeam = XMLUtils.parseBoolean(shopsRoot.getAttribute("cross-team"), false);
+        boolean persistent = XMLUtils.parseBoolean(shopsRoot.getAttribute("persistent"), false);
+        this.tracker = new SimplePurchaseTracker(crossTeam, persistent);
+
+        for (Element shop : shopsRoot.getChild("shops").getChildren()) {
+            context.features().define(shop, parseShop(shop));
+        }
+    }
+
+    private PurchasableSet parseItemSet(Element el) throws InvalidXMLException {
+        Set<Purchasable> purchasables = Sets.newHashSet();
+        for (Element element : el.getChildren()) {
+            purchasables.add(new PurchasableKit(new ItemCreator(XMLUtils.parseMaterial(Node.fromAttr(element, "icon"))),
+                    integerParser.parse(Node.fromRequiredAttr(element, "slot")),
+                    doubleParser.parse(Node.fromRequiredAttr(element, "price")),
+                    context.features().reference(Node.fromRequiredAttr(element, "currency"), Currency.class),
+                    typeParser.parse(Node.fromRequiredAttr(element, "scope")),
+                    XMLUtils.parseBoolean(Node.fromRequiredAttr(element, "incremental"), false),
+                    filterParser.property(element, element.getAttribute("purchase-filter").getValue()).optional(new StaticFilter(Filter.QueryResponse.ALLOW)),
+                    kitParser.property(element, "kit").required()));
+        }
+        return new PurchasableSet.Impl(purchasables);
+    }
+
+    private Currency parseCurrency(Element el) throws InvalidXMLException {
+        Node node = new Node(el);
+        switch (el.getName()) {
+            case "material":
+                Material material = XMLUtils.parseMaterial(node);
+                Node nameSingle = Node.fromAttr(el, "name-singular");
+                Node namePlural = Node.fromAttr(el, "name-plural");
+                BaseComponent name = new TranslatableComponent(
+                        translator.materialKey(material)
+                                .orElseThrow(() -> new InvalidXMLException("No localized name for material " + material))
+                );
+                BaseComponent single = nameSingle == null ? name : new TextComponent(nameSingle.getValue());
+                BaseComponent plural = namePlural == null ? name : new TextComponent(namePlural.getValue());
+                return new MaterialCurrency(material, single, plural, doubleParser.parse(Node.fromAttr(el, "value")));
+            case "exp":
+                return new ExperienceCurrency();
+            default:
+                throw new InvalidXMLException("Specified currency type not found.", el);
+        }
+    }
+
+    private Shop parseShop(Element el) throws InvalidXMLException {
+        final Node node = new Node(el);
+
+        final String title = el.getAttributeValue("title", "Shop");
+
+        final int rows = integerParser.parse(Node.fromAttr(el, "rows"));
```

Comment:
So you can choose the size of the shop and therefore choose where your items are relatively placed.